realm/realm-library/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -97,6 +97,9 @@ public DynamicRealmObject(RealmModel obj) {
         proxyState.getRealm$realm().checkIfValid();
 
         long columnKey = proxyState.getRow$realm().getColumnKey(fieldName);
+        if (proxyState.getRow$realm().isNull(columnKey)) {
```

Comment:
This seems to impact more than just `null` values, so we should have unit tests for all the data types.

Reply:
Implemented tests for `null` values for all types, but I would assume all-non values to be sufficiently test by existing tests. Or what are you missing specifically?

Reply:
No, those tests should be fine